plugins/woocommerce/tests/e2e-pw/global-setup.js

```diff
@@ -166,4 +166,13 @@ module.exports = async ( config ) => {
 	await adminContext.close();
 	await customerContext.close();
 	await browser.close();
+
+	if ( process.env.RESET_SITE === 'true' ) {
+		const { site } = require( './utils' );
+
+		await site.reset(
+			process.env.CONSUMER_KEY,
+			process.env.CONSUMER_SECRET
+		);
+	}
```

Comment:
One source of flakiness, especially in the shopper side of the E2E tests, are the pre-existing test data in our nightly smoke test site. Clearing them during the global setup phase helped a lot in making the tests more stable, so I added these functions here.

This wipe-out is behind a `RESET_SITE` toggle as mentioned [here](https://github.com/woocommerce/woocommerce/pull/35760#discussion_r1041812888) so that users of our E2E tests will have an option to wipe out all these pre-existing data or not.